src/nvim/path.c

```diff
@@ -329,20 +329,25 @@ int vim_fnamencmp(char_u *x, char_u *y, size_t len)
 #endif
 }
 
-/*
- * Concatenate file names fname1 and fname2 into allocated memory.
- * Only add a '/' or '\\' when 'sep' is TRUE and it is necessary.
- */
-char_u *concat_fnames(char_u *fname1, char_u *fname2, int sep)
+/// Concatenate file names fname1 and fname2 into allocated memory.
+/// 
+/// Only add a '/' or '\\' when 'sep' is TRUE and it is necessary.
+///
+/// @param fname1 is the first part of the path or filename
+/// @param fname2 is the second half of the path or filename
+/// @param sep    is a flag to indicate a path separator should be added
+//                if necessary
+/// @returns pointer to newly allocated concatenated filename
+char *concat_fnames(const char *fname1, const char *fname2, bool sep)
   FUNC_ATTR_NONNULL_RET
```

Comment:
fname1 and fname2 cannot be null. So, this can be:
```c
char *concat_fnames(const char *fname1, const char *fname2, bool sep)
  FUNC_ATTR_NONNULL_ARG(1, 2) FUNC_ATTR_NONNULL_RET
```